gnosis-imap/gnosis-imap-spec.ini

```diff
@@ -148,6 +148,158 @@ proxy_storage:
+pc: 0 => _
+gas: #gas(STARTGAS, 0, 0) => _
+memoryUsed: 0 => _
+refund: _ => _
+coinbase: _ => _
+proxy_storage:
+    store(store(store(store(M1:IMap, #hashedLocation({COMPILER}, {OWNERS}, OWNER), INIT_OWNER),
+                                     #hashedLocation({COMPILER}, {OWNERS}, PREV_OWNER), INIT_PREV_OWNER),
+                                     #hashedLocation({COMPILER}, {THRESHOLD}, .IntList), THRESHOLD),
+                                     #hashedLocation({COMPILER}, {OWNERCOUNT}, .IntList), OWNERCOUNT)
+    => M2:IMap
++requires:
+    ; Address public constant SENTINEL_OWNERS = address(0x1);
+    andBool SENTINEL ==Int 1
+    ; Range
+    andBool #rangeAddress(  PREV_OWNER)
+    andBool #rangeAddress(  OWNER)
+    andBool #rangeAddress(  INIT_OWNER)
+    andBool #rangeAddress(  INIT_PREV_OWNER)
+    andBool #rangeUInt(256, NEW_THRESHOLD)
+    andBool #rangeUInt(256, THRESHOLD)  
+    andBool #rangeUInt(256, OWNERCOUNT)
+    andBool #rangeUInt(256, OWNERCOUNT -Int 1)
+    ; Requirements from code
+    ; Only allow to remove an owner, if threshold can still be reached.
+    andBool OWNERCOUNT -Int 1 >=Int NEW_THRESHOLD
+    ; Validate owner address and check that it corresponds to owner index
+    andBool OWNER =/=Int 0
+    andBool OWNER =/=Int SENTINEL
+    andBool INIT_PREV_OWNER ==Int OWNER
+    ; path condition
+    andBool OWNER =/=Int PREV_OWNER     // implied from 'owners[prevOwner] == owner' and 'owner != SENTINEL_OWNERS'
+    ; property to verify
+    andBool #PROXY_ID ==Int MSG_SENDER
+
+[removeOwner-success-1]
+log: _:List ( .List => ListItem(#abiEventLog(#PROXY_ID, "RemovedOwner", #address(OWNER))) )
++requires:
+    ; Requirements from code
+    ; Threshold is unchanged.
+    andBool THRESHOLD ==Int NEW_THRESHOLD 
+ensures:
+   ensures true
+   andBool select(M2, #hashedLocation({COMPILER}, {OWNERS}, OWNER))         ==Int 0
+   andBool select(M2, #hashedLocation({COMPILER}, {OWNERS}, PREV_OWNER))    ==Int INIT_OWNER
+   andBool select(M2, #hashedLocation({COMPILER}, {THRESHOLD}, .IntList))   ==Int THRESHOLD
+   andBool select(M2, #hashedLocation({COMPILER}, {OWNERCOUNT}, .IntList))  ==Int OWNERCOUNT -Int 1 
+   andBool M1 ==IMap M2 except (
+        SetItem(#hashedLocation({COMPILER}, {OWNERS}, OWNER))
+        SetItem(#hashedLocation({COMPILER}, {OWNERS}, PREV_OWNER))
+        SetItem(#hashedLocation({COMPILER}, {THRESHOLD}, .IntList))
+        SetItem(#hashedLocation({COMPILER}, {OWNERCOUNT}, .IntList)) 
+        .Set)
+
+[removeOwner-success-2]
+log: _:List ( .List => ListItem(#abiEventLog(#PROXY_ID, "RemovedOwner", #address(OWNER))) 
+                       ListItem(#abiEventLog(#PROXY_ID, "ChangedThreshold", #uint256(NEW_THRESHOLD))) )
++requires:
+    ; Requirements from code
+    ; Threshold is changed.
+    andBool THRESHOLD =/=Int NEW_THRESHOLD
+    ; Requirements from changeThreshold
+    ; Validate that threshold is smaller than number of owners
+    andBool NEW_THRESHOLD <=Int OWNERCOUNT 
```

Comment:
At this point, the condition `_threshold <= ownerCount` denotes `NEW_THRESHOLD <=Int OWNERCOUNT -Int 1`.

But we already have `OWNERCOUNT -Int 1 >=Int NEW_THRESHOLD` in the pre-condition, so you don't need this at all.